<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgPrefRestServerDlg</class>
 <widget class="QWidget" name="DlgPrefRestServerDlg">
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QCheckBox" name="checkBoxEnableRestServer">
     <property name="text">
      <string>Enable REST API</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxNetwork">
     <property name="title">
      <string>Network</string>
     </property>
     <layout class="QFormLayout" name="formLayoutNetwork">
      <item row="0" column="0">
       <widget class="QLabel" name="labelHost">
        <property name="text">
         <string>Host:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="lineEditHost"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelPort">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBoxPort"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxAuthentication">
     <property name="title">
      <string>Authentication</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayoutAuth">
      <item>
       <layout class="QFormLayout" name="formLayoutAuth">
        <item row="0" column="0">
         <widget class="QLabel" name="labelToken">
          <property name="text">
           <string>Token:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEditAuthToken"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayoutAuthWarning">
        <item>
         <widget class="QLabel" name="labelAuthWarningIcon">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelAuthWarning">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Basic authentication without TLS will expose your credentials. Enable HTTPS to protect the REST API.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxTls">
     <property name="title">
      <string>TLS</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayoutTls">
      <item>
       <widget class="QCheckBox" name="checkBoxUseHttps">
        <property name="text">
         <string>Use HTTPS</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxAutoGenerateCertificate">
        <property name="text">
         <string>Automatically generate self-signed certificate</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayoutTls">
        <item row="0" column="0">
         <widget class="QLabel" name="labelCertPath">
          <property name="text">
           <string>Certificate path:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayoutCert">
          <item>
           <widget class="QLineEdit" name="lineEditCertPath"/>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonBrowseCert">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelKeyPath">
          <property name="text">
           <string>Private key path:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayoutKey">
          <item>
           <widget class="QLineEdit" name="lineEditKeyPath"/>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonBrowseKey">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxRequireTls">
        <property name="text">
         <string>Require TLS for control routes</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayoutTlsStatus">
        <item>
         <widget class="QLabel" name="labelTlsStatusIcon">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelTlsStatus">
          <property name="visible">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayoutStatus">
     <item>
      <widget class="QLabel" name="labelStatusIcon">
       <property name="visible">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelStatus">
       <property name="visible">
        <bool>false</bool>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="wordWrap">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
